src/ui/view/share/data/dropDown/dropDown.tsx

```diff
@@ -0,0 +1,157 @@
+import type { FC } from 'react'
+import { useMemo, useCallback, useState, useEffect } from 'react'
+import { Collapsible } from '@/ui/component/collapsible/collapsible'
+import { SearchBar } from '@/ui/component/searchbar/searchbar'
+import { isSubstringOf } from '@/util/util'
+import { DynamicCheckbox } from '@/ui/view/dataverse/component/dynamicCheckbox/dynamicCheckbox'
+import { NoResultFound } from '@/ui/view/dataverse/component/noResultFound/noResultFound'
+import { Tag } from '../../../../component/tag/tag'
+import './dropDown.scss'
+import { DROPDOWN_MAX_SEARCH_RESULTS, DROPDOWN_TAGS_DISPLAY_LIMIT } from '@/ui/config'
+
+export type DropDownOption = {
+  value: string
+  label: string
+  disabled?: boolean
+  selected?: boolean
+}
+
+type SelectionItemType = 'checkbox'
+
+type DropDownProps = {
+  filterName: string
+  filterValues: string[]
+  searchPlaceholder: string
+  selectionType: SelectionItemType
+  maxSearchResults?: number
+  onSelectionChange: (values: string[]) => string[]
```

Comment:
Callback function should return nothing.
```suggestion
  onSelectionChange: (value: string[]) => void
```